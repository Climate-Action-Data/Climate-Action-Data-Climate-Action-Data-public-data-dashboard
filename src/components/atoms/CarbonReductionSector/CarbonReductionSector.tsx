import { Box, ColorProps, Flex, Text } from "@chakra-ui/react"
import { SectorPieChart } from "../SectorPieChart/SectorPieChart"

interface CarbonReductionSectorProps {
    colorChart: string[],
    data: { value: number, label: string }[]
}


export const CarbonReductionSector = ({ colorChart, data }: CarbonReductionSectorProps): React.JSX.Element => {
    return <Flex>
        <Flex height="150px" width="200px" padding={2}><SectorPieChart data={data} colorChart={colorChart} /></Flex>
        <Flex flexDirection={"column"} justifyContent={"space-evenly"} flex={1}>
            {data.map((sector, idx) => (
                <Box><Text fontWeight="500" color={colorChart[idx]}>{sector.value}% {sector.label}</Text></Box>
            ))}
        </Flex>
    </Flex>
}