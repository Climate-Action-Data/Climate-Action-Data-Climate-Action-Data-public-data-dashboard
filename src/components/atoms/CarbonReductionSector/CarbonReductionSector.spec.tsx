import { render } from '@testing-library/react'
import { CarbonReductionSector } from './CarbonReductionSector';
import React from 'react';

it('renders correctly', () => {
    const { container } = render(<CarbonReductionSector
        colorChart={["green.600", "green.700", "green.800", "green.900"]}
        data={[
            { label: "Renewable Energy", value: 40 },
            { label: "Waste Disposal", value: 24 },
            { label: "Energy Efficiency", value: 19 },
            { label: "Others", value: 17 }]}
    />)
    expect(container).toMatchSnapshot();
});