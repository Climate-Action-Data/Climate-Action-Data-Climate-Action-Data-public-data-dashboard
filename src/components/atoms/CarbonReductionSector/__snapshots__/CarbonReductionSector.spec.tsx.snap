// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <div
      class="css-1775nv6"
    >
      <div
        style="width: 100%; height: 100%;"
      />
    </div>
    <div
      class="css-51blsq"
    >
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-1tjw2bj"
        >
          40
          % 
          Renewable Energy
        </p>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-1pv6nyr"
        >
          24
          % 
          Waste Disposal
        </p>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-ma7g46"
        >
          19
          % 
          Energy Efficiency
        </p>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-4206h5"
        >
          17
          % 
          Others
        </p>
      </div>
    </div>
  </div>
</div>
`;
